runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -44,8 +46,8 @@ public Layout getLayout() {
      *
      * @return The number of nodes required for a quorum.
      */
-    public int getQuorumNumber() {
-        return (int) (getCurrentLayout().getLayoutClientStream().count() / 2) + 1;
+    public int getQuorumNumber(Layout layout) {
```

Comment:
This is not correct. You cannot rely on the Layout from management server. You have the epoch in the current layout and epoch you want to move to. You can do a precondition check before moving forward. We discussed this better in person.